feat(cart): 장바구니 기능 구현

3uomlkh opened the pull request:
## 변경 사항
 - 기능 추가 : `addCartItem`, `getCart`, `removeCartItem`, `removeAllCartItem`
 - 도메인 연계 : `User` 도메인과 연동

## 구현 내용
- 장바구니에 메뉴 추가, 조회, 삭제 기능 구현
- 같은 사용자가 이미 담은 메뉴를 담으면 수량만 증가하도록 구현
- 마지막 업데이트날부터 하루가 지나면 장바구니가 비워지도록 구현

## 추가해야하는 사항 
- `Menu` 도메인과 연동 필요
- 세션 방식에서 JWT로 변경 필요

Closes #5

3uomlkh committed:
feat(Cart): 장바구니 추가 기능 구현 #5

### 장바구니 추가 기능 구현
- 세션 사용해 사용자 구분, 추후 JWT로 변경 예정
- 같은 사용자가 같은 메뉴 추가 시 수량만 변경
- 현재 Menu 기능 개발 전이므로 임시 필드로 MenuId 사용

Signed-off-by: 3uomlkh <3uomlkh@gmail.com>

3uomlkh committed:
Merge branch 'develop' into feat/cart

3uomlkh committed:
feat(Cart): 장바구니 조회 기능 구현 #5

### 장바구니 조회 기능 구현
- 사용자가 추가한 메뉴 모두 조회
- 현재 Menu 기능 개발 전이므로 임시 필드로 MenuId 사용

Signed-off-by: 3uomlkh <3uomlkh@gmail.com>

3uomlkh committed:
feat(Cart): 장바구니 삭제 기능 구현 #5

### 장바구니 삭제 기능 구현
- 단일 메뉴 삭제 기능
- 장바구니 전체 비우기 기능
- 장바구니 hard delete 적용
- 현재 Menu 기능 개발 전이므로 임시 필드로 MenuId 사용

Signed-off-by: 3uomlkh <3uomlkh@gmail.com>

3uomlkh committed:
feat(Cart): 장바구니 에러 메세지 상수화 #5

### ErrorCartMessage Enum 클래스 생성
- 사용자 관련 메세지는 추후 User에서 가져오는 것으로 변경 예정
- 예외 처리 추가될 경우 메세지 추가 예정

Signed-off-by: 3uomlkh <3uomlkh@gmail.com>

3uomlkh committed:
feat(Cart): 하루 지난 장바구니 삭제 기능 구현 #5

### 마지막 업데이트일로부터 하루 지난 장바구니 삭제 기능
- 매일 새벽 3시 스케줄러 실행
- 가장 최근 추가된 메뉴의 날짜가 하루가 지났을 시 장바구니 전체 비우기

Signed-off-by: 3uomlkh <3uomlkh@gmail.com>

3uomlkh committed:
merge: 충돌 해결 후 병합 완료

mixedsider commented on src/main/java/xyz/tomorrowlearncamp/outsourcing/OutsourcingApplication.java:
```diff
@@ -2,11 +2,12 @@
 import static org.springframework.data.web.config.EnableSpringDataWebSupport.PageSerializationMode.VIA_DTO;
 
 @SpringBootApplication
+@EnableScheduling
```
컨피그 파일 따로 만들어서 올려주세요.

mixedsider commented on src/main/java/xyz/tomorrowlearncamp/outsourcing/domain/cart/controller/UserCartController.java:
```diff
@@ -0,0 +1,51 @@
+    @PostMapping
+    public ResponseEntity<AddCartResponseDto> addCartItem(
+            @SessionAttribute(name = "LOGIN_USER") Long userId,
+            @RequestBody AddToCartRequestDto dto
```
@Valid 넣어주세요.

mixedsider commented on src/main/java/xyz/tomorrowlearncamp/outsourcing/domain/cart/entity/CartEntity.java:
```diff
@@ -0,0 +1,42 @@
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    Long id;
```
private 빠졌습니다.

mixedsider commented on src/main/java/xyz/tomorrowlearncamp/outsourcing/domain/cart/dto/request/AddToCartRequestDto.java:
```diff
@@ -0,0 +1,14 @@
+@Getter
+public class AddToCartRequestDto {
+
+    @NotBlank
```
@NotBlank 동작 확인해주세요.

mixedsider reviewed (changes requested):
review: 코드 검토 완료

- 부분적으로 수정하시고 하면 끝날듯합니다.

3uomlkh committed:
feat(Cart): 장바구니 피드백 반영 수정 #5

### 피드백 반영 수정 내용
- private 접근제한자 추가
- SchedulingConfig 클래스 생성 후 EnableScheduling 분리
- NotBlank를 NotNull과 Positive로 수정
- Valid 어노테이션 추가

Signed-off-by: 3uomlkh <3uomlkh@gmail.com>

mixedsider reviewed (approved):
reivew : 코드 검토 완료

3uomlkh committed:
feat(Cart): 장바구니 피드백 반영 수정 #5

### 피드백 반영 수정 내용
- AddToCartRequestDto에 private 접근제한자 추가

Signed-off-by: 3uomlkh <3uomlkh@gmail.com>

mhoo999 reviewed (approved):
문제없어 보입니다.
